proposals/top-level-members.md

```diff
@@ -0,0 +1,117 @@
+# Top-Level Members
+
+Champion issue: (TODO)
+
+## Summary
+
+Allow some members (methods, operators, extension blocks, and fields) to be declared in namespaces
+and make them available when the corresponding namespace is imported.
+
+```cs
+// util.cs
+namespace MyApp;
+
+void Print(string s) => Console.WriteLine(s);
+
+string Capitalize(this string input) =>
+    input.Length == 0 ? input : char.ToUpper(input[0]) + input[1..];
+```
+
+```cs
+// app.cs
+#!/usr/bin/env dotnet
+
+using MyApp;
+
+Print($"Hello, {args[0].Capitalize()}!");
+```
+
+```cs
+// Fields are useful:
+namespace MyUtils;
+
+static string? cache;
```

Comment:
If all members are `static` is the modifier necessary here?  Without it would that pose an issue with the compiler correctly identifying that `cache` should be a field vs. a local variable in an implicit entry point?

Reply:
No, good point, that should not be necessary, will update.

Reply:
The proposal actually already disallows `static` modifier, I just got it wrong in the examples.